examples/bank/bank.go

```diff
@@ -117,43 +122,36 @@ func (bank *Bank) continuousMoneyTransfer(cash int64) {
 			if err := runner.Run(batchRead); err != nil {
 				return err
 			}
-			if batchRead.Results[0].Err != nil {
-				return batchRead.Results[0].Err
-			}
+			var err error
 			// Read from value.
 			fromAccount := &Account{}
-			err := fromAccount.decode(batchRead.Results[0].Rows[0].ValueBytes())
-			if err != nil {
-				return err
-			}
-			// Ensure there is enough cash.
-			if fromAccount.Balance < exchangeAmount {
-				return nil
-			}
+			fromAccount.decode(batchRead.Results[0].Rows[0].ValueBytes(), &err)
 			// Read to value.
 			toAccount := &Account{}
-			errRead := toAccount.decode(batchRead.Results[0].Rows[1].ValueBytes())
-			if errRead != nil {
-				return errRead
-			}
+			toAccount.decode(batchRead.Results[0].Rows[1].ValueBytes(), &err)
 			// Update both accounts.
-			batchWrite := &client.Batch{}
 			fromAccount.Balance -= exchangeAmount
 			toAccount.Balance += exchangeAmount
-			if fromValue, err := fromAccount.encode(); err != nil {
-				return err
-			} else if toValue, err := toAccount.encode(); err != nil {
+			fromValue := fromAccount.encode(&err)
```

Comment:
The code assumes that you can overwrite errors with new errors

On Mon, May 18, 2015 at 3:24 PM MBerhault <notifications@github.com> wrote:

> In examples/bank/bank.go
> <https://github.com/cockroachdb/cockroach/pull/1048#discussion_r30539100>:
>
> >  			fromAccount.Balance -= exchangeAmount
> >  			toAccount.Balance += exchangeAmount
> > -			if fromValue, err := fromAccount.encode(); err != nil {
> > -				return err
> > -			} else if toValue, err := toAccount.encode(); err != nil {
> > +			fromValue := fromAccount.encode(&err)
>
> eg: fromAccount may be the result of a failed "decode", but we're still
> going anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1048/files#r30539100>.
>